src/hooks/useHasPassword.js

```diff
@@ -0,0 +1,42 @@
+import { useEffect, useState } from 'react'
+
+import { useClient } from 'cozy-client'
+import { hasPassword as computeHasPassword } from 'cozy-client/dist/models/user'
+
+import { SETTINGS_DOCTYPE } from 'doctypes'
+
+export const useHasPassword = () => {
+  const client = useClient()
+
+  const [hasPassword, setHasPassword] = useState()
+
+  const handlePassphraseCreated = settings => {
+    if (settings._id === 'io.cozy.settings.passphrase') {
+      setHasPassword(true)
+    }
+  }
+
+  useEffect(() => {
+    client.plugins.realtime.subscribe(
+      'created',
```

Comment:
Is this always `created` here? My comprehension was that `io.cozy.settings.passphrase` was always existing but it was just random for OIDC and MagicLink users, so defining a password would consist of replacing the random unknown password by a known one. With that in mind, shouldn't it be `updated`?